strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/BrowseController.java

```diff
@@ -63,13 +79,21 @@
     public ResponseEntity<String> browse()
     {
         try
-        {
-            Map<String,SortedSet<String>> storages = new HashMap<>();
-            storages.put("storages", new TreeSet<String>(configurationManager.getConfiguration()
-                    .getStorages()
-                    .keySet()));
+        {               
+            Set<Entry<String, Storage>> storages = configurationManager.getConfiguration().getStorages().entrySet();
+                                                    
+            List<Path> storagePaths = new ArrayList<Path>();
+            
+            for(Entry<String, Storage> entry : storages)
+            {
+                Storage storage = entry.getValue();
+                String storageBaseDir = storage.getBasedir();
+                storagePaths.add(Paths.get(storageBaseDir));
```

Comment:
why you are using `Paths` here again? we agreed that storages and repositories should be fetched only from configuration.

Reply:
Well I have fetched storages and repositories from configuration . Then only used Paths to get the lastmodified and size from file system for ONLY THOSE repositories and storages.

Reply:
as I see you fetch storage content from FS here, and this is wrong
you should list repositories which defined in configuration instead

Reply:
oh, I see now what you mean 
the thing is that `fetchDirectoryContent` method name confusing a little :)

Reply:
also, @carlspring do we need size attribute calculated for directories? in general do we need any attributes for the directories?

Reply:
@sbespalov  @carlspring  Yes I was going to mention the  same for 'size' attribute . The way I have seen on some similar listings online, they usually have a dash ( - ) in place of size for directories.

Reply:
@sanket407 : Yes, this seems right. You can ahead and change it.

Reply:
@carlspring  Done !

Reply:
@sbespalov : Sorry, I've just realized that this is actually a very bad idea. Every time you browse, the size of the directories will have to be re-calculated. The more your repositories grow, the slower the browsing will be.
@sanket407 : I'm very sorry for realizing this so late, but would you mind reverting this particular change?